code/datums/components/magnetic_catch.dm

```diff
@@ -0,0 +1,23 @@
+/datum/component/magnetic_catch/Initialize()
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+	if(ismovableatom(parent))
+		RegisterSignal(COMSIG_MOVABLE_UNCROSS, .proc/uncross_react)
+	else
+		RegisterSignal(COMSIG_ATOM_EXIT, .proc/exit_react)
+	RegisterSignal(COMSIG_PARENT_EXAMINE, .proc/examine)
+
+/datum/component/magnetic_catch/proc/uncross_react(atom/movable/thing)
+	if(!thing.throwing || thing.throwing.thrower)
+		return
+	qdel(thing.throwing)
```

Comment:
shouldn't you use finalize instead? this skips throwing callbacks

Reply:
I did that intentionally because this is basically stopping the throw from having any effects. I'd rather be preventing the throw in the first place but I couldn't decide on a good way to do that from the component.